Userland/Libraries/LibIMAP/Client.cpp

```diff
@@ -460,4 +502,10 @@ void Client::close()
         m_socket->close();
     }
 }
+
+bool Client::is_open()
+{
+    return m_tls ? m_tls_socket->is_open() : m_socket->is_open();
```

Comment:
To be honest, this looks like it should be a `Variant<TLS::TLSv12, Core::Stream::Socket>`.

Reply:
Well, it should really just be a `NonnullOwnPtr<Core::Stream::Socket>`, but `TLS::TLSv12` isn't a `Core::Stream::Socket` subclass yet. In the future it should all be transparent, so this is just a transitionary period until that becomes possible.